pyof/utils.py

```diff
@@ -0,0 +1,37 @@
+"""General Unpack utils for python-openflow."""
+import pyof.v0x01.common.header
+import pyof.v0x01.common.utils
+import pyof.v0x04.common.header
+import pyof.v0x04.common.utils
+from pyof.foundation.exceptions import UnpackException
+
+pyof_version_libs = {0x01: pyof.v0x01,
+                     0x04: pyof.v0x04}
+
+
+def unpack(packet):
+    """Unpack the OpenFlow Packet and returns a message."""
+    try:
+        version = packet[0]
+    except IndexError:
+        raise UnpackException('null packet')
+
+    try:
+        pyof_lib = pyof_version_libs[version]
+    except KeyError:
+        raise UnpackException('Version not supported')
+
+    try:
+        header = pyof_lib.common.header.Header()
+        header.unpack(packet[:8])
+        message = pyof_lib.common.utils.new_message_from_header(header)
+        binary_data = packet[8:]
+        if binary_data:
+            if len(binary_data) == header.length - 8:
+                message.unpack(binary_data)
+            else:
+                raise UnpackException(
+                    'packet size does not match packet length field')
```

Comment:
@diraol, @erickvermot told me something before that you might not be aware (see my review comment to write it in the module docstring): the purpose of this PR is to provide the same unpack of of_core to python-openflow tests. Thus, my review is based on copying the unpack method. Maybe we should move discussions about changing the current of_code code to another issue, wherever it will be (we must remove this duplication).

Regarding Erick's 2 bullets, I see those issues in a different way, respectively:
- If there's no data to unpack, it's dangerous to assume that it will always mean nothing. It should be up to each message to decide it. Thus, I believe we'd better always call _unpack_.
- This should be done in the message validation. If it is not done yet, I see no reason on implementing it quick and dirty in the wrong place.